MapboxNavigation/RatingControl.swift

```diff
@@ -0,0 +1,141 @@
+            updateSelectionStates()
+        }
+    }
+    @objc dynamic public var normalColor: UIColor = #colorLiteral(red: 0.8508961797, green: 0.8510394692, blue: 0.850877285, alpha: 1) {
+        didSet {
+            updateSelectionStates()
+        }
+    }
+    
+    @objc dynamic public var starSize: CGSize = defaultSize {
+        didSet {
+            configureStars()
+        }
+    }
+    
+    @objc dynamic public var starCount: Int = 5 {
+        didSet {
+            configureStars()
+        }
+    }
+    
+    //MARK: Initializers
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        configureStars()
+    }
+    
+    required public init(coder: NSCoder) {
+        super.init(coder: coder)
+        configureStars()
+    }
+    
+    //MARK: Private Functions
+    private func configureStars() {
+        removeStars()
+        addStars()
+        updateSelectionStates()
+    }
+    
+    private func addStars() {
+        for index in 0..<starCount {
+            let button = UIButton(type: .custom)
+            button.setImage(starTemplate, for: .normal)
+            button.adjustsImageWhenHighlighted = false
+            addButtonSizeConstraints(to: button)
+            
+            let setRatingNumber = NSNumber(value: index + 1)
+            let setRatingString = NumberFormatter.localizedString(from: setRatingNumber, number: .none)
+            let localizedStarLabel = NSLocalizedString("Set %@ star rating", comment: "Accessibility Star Label")
+            button.accessibilityLabel = String.localizedStringWithFormat(localizedStarLabel, setRatingString)
+                
+            
+            button.addTarget(self, action: #selector(RatingControl.ratingButtonTapped(button:)), for: .touchUpInside)
+            
+            addArrangedSubview(button)
+            
+            stars.append(button)
+        }
+    }
+    
+    private func removeStars() {
+        stars.forEach {
+            removeArrangedSubview($0)
+            $0.removeFromSuperview()
+        }
+        stars.removeAll()
+    }
+    
+    private func updateSelectionStates() {
+        for (index, button) in stars.enumerated() {
+            
+            let selected = index < rating
+            button.tintColor = selected  ? selectedColor : normalColor
+            button.isSelected = selected
+            
+            setAccessibility(for: button, at: index)
+        }
+    }
+    
+    private func setAccessibility(for button: UIButton, at index: Int) {
+        setAccessibilityHint(for: button, at: index)
+        
+        let value: String
+        
+        switch rating {
+        case 0:
+            value = NSLocalizedString("No rating set.", comment: "No Rating Set")
+        case 1:
+            value = NSLocalizedString("1 star set.", comment: "One Star Set")
+        default:
+            value = "\(rating) " + NSLocalizedString("stars set.", comment:"(multiple) Stars Set, as in '30 stars set'")
```

Comment:
Use `String.localizedStringWithFormat` in combination with `NSLocalizedString` to support a more generic format. Also, we must specify `bundle:` when using `NSLocalizedString`, otherwise it will look in the main app bundle for a string.

Reply:
- [ ] Better String Localization